tests/operator_test.go

```diff
@@ -874,6 +874,42 @@ spec:
 		})
 	})
 
+	Describe("should reconcile components", func() {
+
```

Comment:
one thing I'm a little paranoid about is the operator getting into an update storm of some sort, where it's fast looping writing objects because it thinks they are out of date for some reason.

What would you think about writing some func test logic that verifies that the new generation list doesn't change after an update is complete. Maybe this could be added to the `waitForKvWithTimeout` function, something that just observes that no changes occur to the generation list for something like 15 seconds after the kv object looks like it's stable.

Reply:
Yeah, that makes sense. I've added a `Consistently` check at the end of this func test to check that the generation stays the same on the KV resource from the time the resource is updated back to it's original state.